addons/crm/tests/test_crm_lead.py

```diff
@@ -119,6 +119,22 @@ def test_crm_lead_creation_partner(self):
         self.assertEqual(lead.zip, self.contact_1.zip)
         self.assertEqual(lead.country_id, self.contact_1.country_id)
 
+    @users('user_sales_manager')
+    def test_crm_lead_suggested_recipient_partner_no_email(self):
+      lead = self.env['crm.lead'].create({
+        'name': 'Test Lead',
+        'partner_id': self.contact_without_email.id,
+        'email_cc': 'test_cc@odoo.com'
+      })
+      result_no_email = lead._message_get_suggested_recipients()
+
+      lead.write({
+        'partner_id': self.contact_2.id
+      })
+      result_with_email = lead._message_get_suggested_recipients()
+      self.assertEqual(result_no_email[lead.id][1][1], self.contact_without_email.name)
+      self.assertEqual(result_with_email[lead.id][1][1], self.contact_2.email)
```

Comment:
Maybe we can just compare to a whole dict, might be clearer what the result is, since there's no other test on this in crm